### DISCUSSION TITLE
Google Calendar API

### DISCUSSION THREAD

**Discussion by cedrickp32** [elgatosf/elgato#112] [Help]
I'm building myselft a plugin to see my Google Calendar on the stream Deck +. Everything is working great as far as showing dummy events, but can't make the Google Calendar API work. I've use some templates files from Google that is using the OAuth login. It's working fine in the browser, but I'm getting errors when trying to authorize when embedded in the inspector panel. Does anyone know how to solve this ? Or maybe another way of doing it.

![Screenshot 2024-04-09 at 00 30 39](https://github.com/elgatosf/elgato/assets/96031553/985b3dff-d2fe-4a84-9ab0-afd75b5902c1)
![Screenshot 2024-04-09 at 00 30 50](https://github.com/elgatosf/elgato/assets/96031553/af0365e7-67e6-4e84-aba7-5ab21af13443)
![Screenshot 2024-04-09 at 00 30 59](https://github.com/elgatosf/elgato/assets/96031553/8be35cae-d493-45b1-883e-f063b748702c)

**Comment by elgatosystems** [comment]
Hi there,

A while ago I faced a similar problem with a Home Assistant integration for Google Calendar and found the only way to do this was to authorise in the browser and copy the credentials.json and token.json files into the required place. I was surprised it worked, but it did and has done for the past two years.

Have a look at this:
https://github.com/Charry2014/GoogleCalendarReader
It is in Python and very hacky but solves the problem you are having.



From: cedrickp32 ***@***.***>
Date: Tuesday, 9. April 2024 at 06:36
To: elgatosf/elgato ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [Elgato] Google Calendar API (Discussion #112)
I'm building myselft a plugin to see my Google Calendar on the stream Deck +. Everything is working great as far as showing dummy events, but can't make the Google Calendar API work. I've use some templates files from Google that is using the OAuth login. It's working fine in the ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌

[Image removed by sender.]<https://us.report.cybergraph.mimecast.com/alert-details/?dep=n9vCTQFfjGwaIxrLTvOhtg%3D%3DxTbTSK4eQ4%2FnW04iaFWn2oUCK38Os%2FgM7G3vfoWgB6LRY0pC%2F20SEsQjmOxH3O0f0RYBfM3uPoni9KZ1YccQnneIXO2p41aDtFupv94O%2FqtVCGl4ggqllbGGO0e2va9DF3BM65z2K%2B4hMhQwDBNCtX4S0EJglJjfbXBjx9oDDlU45AOLgUzpV5lxXwWMoO3%2Fx85g%2BpF0PpCijXtrYKTGuHTxyatShuqpksx7%2BEsx%2FC5o7KLpsd6f8PhQO3E1N%2BfGU3eACvM97pSggNP4d744X9Dc%2FanNq3bKZ%2Bco1JhVIAFM7BaVC1e1nIW%2B89eoeDxz6EBrfKXyd2BATYtp7zTh9Gy9v6X6qhugk7KZSWkGoolay4wBY1TafArJPu2RKNw99sSEw%2BOGFvbOSSPgS%2FwV5SL9JZe7%2FfMuloNeDU%2Fw9a%2BZ2HlC%2F%2BTNuWsLn0sf00Kq%2Be3YMBd0RhJ7cRpNo0Jhx%2BNzkjFjNDGieI0XNPJ%2B8xl%2BPvOTm%2BOd%2FvK3uw82FrWWplxiP0Gab33bJsLKGZknLDqwwYuAsjkuPHBz8PmE%2BjlZ%2Fj3MB9zByv2mclqVLqvWcF81I3tQSlM%2BrLDea1MnQPSpKdefuWvNFg9xsxcBPp3rI2OSAZS6vvpxDVMoslkgQrRVQM99Dg1WuYJ4lS4eqlboK4ioc9A6w4Zi%2FeJ2LvrIT7LoqUz44OOv2XI9uyKq>

I'm building myselft a plugin to see my Google Calendar on the stream Deck +. Everything is working great as far as showing dummy events, but can't make the Google Calendar API work. I've use some templates files from Google that is using the OAuth login. It's working fine in the browser, but I'm getting errors when trying to authorize when embedded in the inspector panel. Does anyone know how to solve this ? Or maybe another way of doing it.

Screenshot.2024-04-09.at.00.30.39.png (view on web)<https://github.com/elgatosf/elgato/assets/96031553/985b3dff-d2fe-4a84-9ab0-afd75b5902c1>
Screenshot.2024-04-09.at.00.30.50.png (view on web)<https://github.com/elgatosf/elgato/assets/96031553/af0365e7-67e6-4e84-aba7-5ab21af13443>
Screenshot.2024-04-09.at.00.30.59.png (view on web)<https://github.com/elgatosf/elgato/assets/96031553/8be35cae-d493-45b1-883e-f063b748702c>

—
Reply to this email directly, view it on GitHub<https://github.com/orgs/elgatosf/discussions/112>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AAJPPYXH6XTGQKWGNQLQNQLY4NV3XAVCNFSM6AAAAABF524PO2VHI2DSMVQWIX3LMV43ERDJONRXK43TNFXW4OZWGQ4DCMRWGU>.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

This email, including any attachments, may contain information that is privileged or confidential and is intended only for the person or entity to which it is addressed. If you are not the intended recipient, please delete the email and any attachments from any computer system and notify us immediately. Any review, retransmission, dissemination or other use of, or taking any action in reliance on, the information contained in this email by persons or entities other than the intended recipient is prohibited and any and all liability arising therefrom is hereby disclaimed.
